src/prefect/tasks/mysql/mysql.py

```diff
@@ -173,13 +184,36 @@ def run(
                 "The 'fetch' parameter must be one of the following - ('one', 'many', 'all')"
             )
 
+        cursor_types = {
+            "cursor": pymysql.cursors.Cursor,
+            "dictcursor": pymysql.cursors.DictCursor,
+            "sscursor": pymysql.cursors.SSCursor,
+            "ssdictcursor": pymysql.cursors.SSDictCursor,
+        }
+
+        if callable(cursor_type):
+            cursor_class = cursor_type
+        elif isinstance(cursor_type, str):
+            cursor_class = cursor_types.get(cursor_type.lower())
+        else:
+            raise TypeError(
+                f"'cursor_type' must be of type str or a full cursor class, got {type(cursor_type)}"
+            )
```

Comment:
If you do `cursor_class = None` here the TypeError does not need to be duplicated

Reply:
Ah good catch. I'm happy to incorporate if it's not more trouble than it's worth on your side. Additionally, I see a CircleCI failure here:

https://app.circleci.com/pipelines/github/PrefectHQ/prefect/6569/workflows/e5016f4d-91a2-4462-9ab0-9caa0004aafd/jobs/102603

Although, the failure cases seem orthogonal to the files changed in this PR. Is this a concern?

Reply:
It's not more trouble on our side! 

I'm not sure what's up with that test. I would ensure that your forked repo is up to date with `master` and merged into this branch.